mlflow/pyfunc/__init__.py

```diff
@@ -2895,95 +2909,96 @@ def predict(self, context, model_input: List[str], params=None) -> List[str]:
     if mlflow_model is None:
         mlflow_model = Model()
     saved_example = None
-
-    hints = None
-    if signature is not None:
-        if isinstance(python_model, ChatModel):
+    signature_from_type_hints = None
+    if isinstance(python_model, ChatModel):
+        if signature is not None:
             raise MlflowException(
                 "ChatModel subclasses have a standard signature that is set "
                 "automatically. Please remove the `signature` parameter from "
                 "the call to log_model() or save_model().",
                 error_code=INVALID_PARAMETER_VALUE,
             )
-        mlflow_model.signature = signature
+        mlflow_model.signature = ModelSignature(
+            CHAT_MODEL_INPUT_SCHEMA,
+            CHAT_MODEL_OUTPUT_SCHEMA,
+        )
+        # For ChatModel we set default metadata to indicate its task
+        default_metadata = {TASK: _DEFAULT_CHAT_MODEL_METADATA_TASK}
+        mlflow_model.metadata = {**default_metadata, **(mlflow_model.metadata or {})}
```

Comment:
```suggestion
        mlflow_model.metadata = default_metadata | (mlflow_model.metadata or {})
```

ref: https://peps.python.org/pep-0584/